llvm/include/llvm/Object/XCOFFObjectFile.h

```diff
@@ -853,6 +853,8 @@ class xcoff_symbol_iterator : public symbol_iterator {
   xcoff_symbol_iterator(const basic_symbol_iterator &B)
       : symbol_iterator(B) {}
 
+  xcoff_symbol_iterator(const SymbolRef *Symbol) : symbol_iterator(*Symbol) {}
```

Comment:
`xcoff_symbol_iterator(const SymbolRef *Symbol) : symbol_iterator(*Symbol) {}`

change to 

`xcoff_symbol_iterator(const XCOFFSymbolRef *Symbol) : symbol_iterator(*Symbol) {}`

otherwise a pointer of  ELFSymbolRef can also be used to  initiate the xcoff_symbol_iterator.

Reply:
Done